File: 18.09.2023/demo/12.js
function sum(str) { //функция для счета суммы чисел
    var sum = 0;
    for (let i = 0; i < str.length; i++) {
        sum += parseInt(str[i])
    }
    return sum;
}

for (let n = 3; n < 10000; n++) { //цикл для обработки n
  let str = `5${"2".repeat(n)}`;
  //console.log(str)
  while (str.includes("52") || str.includes("2222") || str.includes("1122")) { //обработка числа
    if (str.includes("52")) {
    str = str.replace("52", "11");
      //console.log(str)
    }
    if (str.includes("2222")) {
    str = str.replace("2222", "5");
      //console.log(str);
    }
    if (str.includes("1122")) {
    str = str.replace("1122", "25");
      //console.log(str);
    }
  }
  //console.log(sum(str))
  if (sum(str) == 64) { //проверка на сумму чисел
    console.log(n);
  }
}
console.log('Цикл закончен')

File: 18.09.2023/demo/12.py
def sum(a):
    s = 0
    n = int(a)
    while n:
        s += n % 10
        n //= 10
    return s

for i in range(2, 100000):
        str = '5'+'2'*i;
        #print(str)
        while ('52' in str or '2222' in str or '1122' in str):
            if ('52' in str):
                str = str.replace('52', '11')
                #print(str)
            if ('2222' in str):
                str = str.replace('2222', '5')
                #print(str)
            if ('1122' in str):
                str = str.replace('1122', '25')
                #print(str)
        if sum(str) == 64:
            print(i)
print('Цикл закончен')